openmsupply/application-manager-web-app
# #12 mutation example register

Fixes #9 & #12 

Add folders: 
* src/containers
* src/containers/User

* Moves the `App.tsx` file to containers folder and new folder `User`.

## (Issue 9) Changes: `useQuery`
* Basic setup the `UserState` using the useContext and useReducer hooks in a structured way.

* Add files in `src/containers/User`:
* **UserArea** - container that would have nested components (accessing the same context)
* **UserState** - where the user states are stored and the dispatch function to be used by components
* In the `App.tsx` the `UserArea` is wraped by the `UserProvider` to make the magic.

## (Issue 12) Changes: `useMutation`
* Add example of mutation on component `Register`.
* Add cache management to update the current local fetched list of user after the mutation runs.

## Usage & tests
* Run `yarn dev`
* See the list of existing users on the top (`User Area` component).
* Go to the Register page, add first name, last name, email and select a role.
* Click the submit button to see the submission message (snackbar).
* Next the new added user will be displayed on the top `User Area`.
